pitaya-sharp/NPitaya/src/PitayaCluster.API.cs

```diff
@@ -123,7 +122,7 @@ private static void HandleRpcCallback(IntPtr userData, IntPtr ctx, IntPtr rpc)
                 logKind,
                 Marshal.GetFunctionPointerForDelegate(logFunctionCallback),
                 GCHandle.ToIntPtr(logCtx),
-                pitayaMetrics.Ptr,
+                _metricsReporter.GetPitayaPrt(),
```

Comment:
The PitayaCluster class already has a pointer to the pitaya instance, you don't need this method.

Reply:
This method improves encapsulation. It's not a good practice to export such things as pointers relentlessly. Also, there's the feature flag logic embedded in this method, which is important.

Reply:
Right, but the metrics reporter is not the owner of the Pitaya pointer. The owner is the PitayaCluster class.

Reply:
oh nvm. I was confusing the pointers. This is the raw metrics reporter pointer and not the pitaya instance pointer.